### ISSUE TITLE
Feature request - Reenable Tango support

### ISSUE THREAD

**Issue by lvonasek** [google-ar/arcore-android-sdk#638]
Even that ARCore is doing great progress, there is no real interest from app developers. It is because has really big lack of possible business usecases.

Sadly ARCore is the worst thing which could to mobile AR happen. Supporting both Tango and ARCore would have advantage that developers can target highend market (Tango) and lowend market (ARCore).

Note that Tango support for Oppo R(X)17 Pro and Samsung A9 Star would make mobile AR great again (these devices have all HW necessary for Tango support).

**Comment by lvonasek** [comment]
No chance to open this topic on GoogleAR's internal meeting?

For many people ARCore is just a toy and it is still not too late to reconsider the wrong decision about killing Tango.

**Comment by inio** [comment]
Nope.  Tango is dead.  It suffered from requiring special hardware in every device, making it unavailable for the vast majority of users.  Efforts to get that hardware included in a larger collection of devices were unsuccessful.

**Comment by lvonasek** [comment]
Would be at least possible to opensource the code to enable 3rd parties to make possible to continue there where Google failed?

**Comment by inio** [comment]
The majority of the Tango codebase is present in ARCore, so there's not much left to opensource.  Not sure about the Phab 2 Pro, but the depth camera on the ZenFone AR should be available as an Android camera.  The missing piece there would be the calibration of the depth camera, which you should be able to read via TangoCore fairly easily.  With 1.6 ARCore reports the pose of the COLOR camera as Camera.getPose(), so if you get CAMERA_COLOR_PRIMARY to CAMERA_DEPTH and the DEPTH intrinsics you'll have what you need.

**Comment by lvonasek** [comment]
Thank you for the answer, unfortunaly depth camera is not available using Android Camera API on Asus Zenfone AR.

It can be tested by editing android-ndk-master/camera/basic/src/main/cpp/camera_manager.cpp
by replacing the line with openCamera with this:
  //"0" = rear, "1" = front, "2" = fisheye, "3" = depth
  CALL_MGR(openCamera(cameraMgr_, "3".c_str(), GetDeviceListener(),
                      &cameras_[activeCameraId_].device_));

with "1" it works (front camera) but with "2" or "3" it crashes on this line with message:
Abort message: 'NDKCamera call failed with code: 0xffffd8ef, ACAMERA_ERROR_INVALID_PARAMETER'
by calling that line is in the logcat written:
E/ACameraManager: getCameraCharacteristics: Camera ID 3 does not exist!

Is it needed to do same magic to make camera available for NDKCamera?

**Comment by inio** [comment]
Turns out I had been misinformed.  Just talked to some oldschool Tango folk and you're 100% correct.  Depth camera was accessed through private APIs and loading a vendor-specific library into TangoCore.

**Comment by lvonasek** [comment]
OK, thank you for the confirmation. Also there is no way for us how to read without Tango depth sensor data. We have to wait on #120 which is according to @claywilkinson in #603 not a priority at all.

Would be possible to get some time estimation of depth sensor support on ARCore? 2019? 2020? 2021?

Try to understand that we showed in 2017 to the customers Tango based solutions and when we show them in 2018 solutions based on ARCore they are not interested in it. It is very hard for us when we cannot promise them any time estimation even with accuracy +/- half year.

**Comment by lvonasek** [comment]
Today I lost my job with justification: 
There is no Tango anymore and no another serious mobile AR technology and that's why investment into AR development is not going to happen

**Comment by mpottinger** [comment]
Now in 2019, more phones are including time of flight depth sensors, so it seems Tango was just a little ahead of its time. In order to become really useful beyond some toy apps, ARCore will need to support  these sensors and the more advanced features using depth on the phones that have it.

**Comment by vanboxsel-engineering** [comment]
I don't understand the lack of support for TOF sensors. 
I mean, how is [this](https://www.youtube.com/watch?v=z_seJgR5CCA&feature=youtu.be) not awesome?

**Comment by raju535482** [comment]
Since 2020, Apple is launching every iPhone pro and max model with LiDAR ToF sensors, Now you Google guys have your own pixel brand, so just add in it. in iOS they have added many cool features which we android developers literally have to do lot of work to implement. Apple has full body detection, hand joint-detection, etc. many things.
